src/python/pants/backend/python/rules/repl.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import logging
+from pathlib import Path
```

Comment:
Semi-nit: it looks like `PurePath` would work for everything in this file and I recommend changing to it. `PurePath` is a superclass of `Path` and has all the same methods as `Path`, except for methods that actually inspect the real filesystem. For example, there is no `PurePath.exists()` or `PurePath.is_file()`.

Why use `PurePath`? It ensures we don't accidentally leak into the filesystem when we didn't intend to and it better expresses the intent of the code, that you're solely doing path manipulation and not actually reading from the filesystem.